### ISSUE TITLE
spec: Clarify use of metadata (headers) in clients

### ISSUE THREAD

**Issue by johanbrandhorst** [grpc/grpc-web#337]
Hi!

An issue was raised [1] against `improbable-eng/grpc-web` asking what the best way is for a client to specify the HTTP/2 `:authority` header for routing between services in the backend. The spec mentions a few reserved headers, but I think it would be nice to have clarity on exactly which headers have a special meaning for the gRPC protocol and how they should be specified in a HTTP/1.1 world.

Should the client simply specify the metadata header as `authority: <value>`?

1: https://github.com/improbable-eng/grpc-web/issues/266#issuecomment-432512985

**Comment by stanley-cheung** [comment]
Yes the expectation is that the header is passed as `authority: value`. But this depends on the proxy implementation though - currently I believe Envoy just passes everything through but it may not be the case for another proxy implementation. So we need to keep this in mind when we implement / evaluation another proxy implementation.

**Comment by johanbrandhorst** [comment]
So it may be necessary to add a section to the spec with a list of HTTP/1.1 headers that should be translated to special HTTP2 headers, like `authority` -> `:authority`. Does that make sense?

**Comment by stanley-cheung** [comment]
Sorry for the confusion. When I wrote my previous comment I didn't actually look up the semantics of `:authority` http2 header. I was confused with another header we just added support on: `grpc-timeout` and thought we were dealing with roughly they are the same thing but they are not. I do know that Envoy handles and pass-thru the `grpc-timeout` header properly. I also realize what @johanbrandhorst means by needing some sort of documentations on what header we explicitly support. I must admit that that list is a rather small subset for now. We need to better document and improve on this for sure.
